tests/Aql/SortedCollectExecutorTest.cpp

```diff
@@ -410,287 +503,525 @@ TEST(SortedCollectExecutorTestRowsUpstreamCount, test) {
+    auto engine =
+        std::make_unique<ExecutionEngine>(*fakedQuery, SerializationFormat::SHADOWROWS);
+    fakedQuery->setEngine(engine.release());
+  }
+};
 
-  OutputAqlItemRow result(std::move(block), infos.getOutputRegisters(),
-                          infos.registersToKeep(), infos.registersToClear());
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::HASMORE);
-  ASSERT_TRUE(result.produced());
-  result.advanceRow();
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::HASMORE);
-  ASSERT_TRUE(result.produced());
-  result.advanceRow();
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_TRUE(result.produced());
-  result.advanceRow();
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_FALSE(result.produced());
-
-  std::vector<std::string> myStrings;
-  std::vector<int> myCountNumbers;
-  auto newBlock = result.stealBlock();
-
-  // check for types
-  AqlValue x = newBlock->getValue(0, 1);
-  ASSERT_TRUE(x.isString());
-  EXPECT_EQ(x.slice().copyString(), "a");
-
-  // Check the count register
-  AqlValue c = newBlock->getValue(0, 2);
-  ASSERT_TRUE(c.isNumber());
-  EXPECT_EQ(c.slice().getInt(), 1);
-
-  // check for types
-  x = newBlock->getValue(1, 1);
-  ASSERT_TRUE(x.isString());
-  EXPECT_EQ(x.slice().copyString(), "aa");
-
-  // Check the count register
-  c = newBlock->getValue(1, 2);
-  ASSERT_TRUE(c.isNumber());
-  EXPECT_EQ(c.slice().getInt(), 1);
-
-  // check for types
-  x = newBlock->getValue(2, 1);
-  ASSERT_TRUE(x.isString());
-  EXPECT_EQ(x.slice().copyString(), "aaa");
-
-  // Check the count register
-  c = newBlock->getValue(2, 2);
-  ASSERT_TRUE(c.isNumber());
-  EXPECT_EQ(c.slice().getInt(), 1);
+TEST_P(SortedCollectExecutorTestSplit, split_1) {
+  auto [split] = GetParam();
+
+  ExecutorTestHelper<SortedCollectExecutor>(*fakedQuery)
+      .setInputValueList(1, 1, 1, 2, 3, 4, 4, 5)
+      .setInputSplitType(split)
+      .setCall(AqlCall{2, AqlCall::Infinity{}, 2, true})
+      .expectOutputValueList(3, 4)
+      .expectSkipped(3)
+      .expectedState(ExecutionState::DONE)
+      .run(std::move(infos));
 }
 
-}  // namespace aql
-}  // namespace tests
-}  // namespace arangodb
+TEST_P(SortedCollectExecutorTestSplit, split_2) {
+  auto [split] = GetParam();
+
+  ExecutorTestHelper<SortedCollectExecutor>(*fakedQuery)
+      .setInputValueList(1, 1, 1, 2, 3, 4, 4, 5)
+      .setInputSplitType(split)
+      .setCall(AqlCall{2, 2, AqlCall::Infinity{}, false})
+      .expectOutputValueList(3, 4)
+      .expectSkipped(2)
+      .expectedState(ExecutionState::HASMORE)
+      .run(std::move(infos));
+}
+
+TEST_P(SortedCollectExecutorTestSplit, split_3) {
+  auto [split] = GetParam();
+
+  ExecutorTestHelper<SortedCollectExecutor>(*fakedQuery)
```

Comment:
I like these tests, for my test they are easier to understand than the above tests.
I imagine that we need some more tests here.
e.g.
empty input list => by definition collect should write an empty line than still. (in certain cases)

Also we might need more tests for the possible Aggregates here.

However I am fine with having more tests as TODO and merge this branch earlier, we covered a lot in JS tests already and the testing framework should be merged sooner than later.